### PR TITLE
Added missing param to lambda module to pass it through to boto

### PR DESCRIPTION
**Pull request by stefanhorning**
##### SUMMARY

Added missing `tracing_mode` param that passes value through to `TracingConfig` in boto, which was somehow not implemented in this module so far.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lambda module

##### ADDITIONAL INFORMATION
Module now allows setting all parameters the boto method/AWS API allows.

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/lambda.py
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/lambda.py:99]: Do you mind adding this to the return documentation and updating the aws_lambda test target to use it?

Comment (stefanhorning) [reply lib/ansible/modules/cloud/amazon/lambda.py:99]: Done now

**Comment by stefanhorning** [comment]
Also did some cleanups in the modules test play.

**Comment by stefanhorning** [comment]
@s-hertel tests are green again after adding testcases for my changes (plus doing a bit of cleanup in the test play). So feel free to review again and let me know if more changes are needed, otherwise you can merge if you want.

**Comment by s-hertel** [comment]
@stefanhorning I was going to push to the branch because I didn't want to nitpick and you made changes so quickly, but I don't seem to have permission to do that. Please cherry-pick the commit or implement the things I did here - https://github.com/ansible/ansible/commit/8abd2366f5e3e677ec8bdfb7ef17942ed7372218. I updated the lambda execution role so those tests should pass now.

**Comment by stefanhorning** [comment]
Ok, will pull in your (still relevant) changes

**Comment by s-hertel** [comment]
Thanks! :smile: I think they were all still relevant. I added a test for changed == False. Can you add and also please use module_defaults too instead of the yaml anchor for credentials? We updated the dev guide to recommend it. The ec2_ credentials are old and just available for legacy purposes - I also updated those to use the aws_ ones. After that I'll merge once shippable is green.

File: test/integration/targets/aws_lambda/tasks/main.yml
Comment (s-hertel) [inline test/integration/targets/aws_lambda/tasks/main.yml:142]: Please add the assertion I had above that PassThrough is set as the default before this.

**Comment by stefanhorning** [comment]
Ah, ok didn't spot the difference in that credentials block :)

File: test/integration/targets/aws_lambda/tasks/main.yml
Comment (stefanhorning) [reply test/integration/targets/aws_lambda/tasks/main.yml:142]: Done

File: test/integration/targets/aws_lambda/tasks/main.yml
Comment (s-hertel) [inline test/integration/targets/aws_lambda/tasks/main.yml]: The YAML anchors should also be removed from the file now that it is no longer defined. `module_defaults` sets the default parameters for the modules in the `aws` group so these will all get the credentials and you should only set them on the task if you want to override the defaults. There are a number of tasks that expect `region` to not be set. Those tasks need `region:` added to them like I had in my commit. Sorry, this is why I was just going to push to the branch. Fiddly details.

Comment (stefanhorning) [reply test/integration/targets/aws_lambda/tasks/main.yml]: yeah, my bad. really just should have cherry-picked your commit, thought it would be easier to just copy missing pieces :P

File: test/integration/targets/aws_lambda/tasks/main.yml
Comment (s-hertel) [inline test/integration/targets/aws_lambda/tasks/main.yml:172]: It's still missing the changed == False test above this one. That's how I found the modification bug. Here's what I had:
```
    - name: test no changes are made with the same parameters
      lambda:
        name: "{{lambda_function_name}}"
        runtime: "nodejs10.x"
        tracing_mode: 'Active'
        handler: "mini_lambda.handler"
        role: "ansible_lambda_role"
      register: update_result

    - name: assert that update succeeded
      assert:
        that:
          - update_result is not failed
          - update_result.changed == False
          - update_result.configuration.runtime == 'nodejs10.x'
          - update_result.configuration.tracing_config.mode == 'Active'
```

**Review by s-hertel — COMMENTED** [review]
Sorry, almost have everything. Can you also amend the commit with `@s-hertel` and just use `s-hertel`? That way I don't get pings from it for eternity. (You'll need to --force push to reflect that edit here)

**Comment by stefanhorning** [comment]
Will rebase this branch to squash a few commits/correcting `@` in commit message once test is green

File: test/integration/targets/aws_lambda/tasks/main.yml
Comment (stefanhorning) [reply test/integration/targets/aws_lambda/tasks/main.yml:172]: Also done now

**Review by s-hertel — APPROVED** [review]
Thanks :smile:

**Comment by stefanhorning** [comment]
Yeah, I think we are finally good now, but now I overwhelmed shippable :)

**Pull request merged**
